src/pybind/mgr/cephadm/module.py

```diff
@@ -1649,7 +1649,8 @@ def _daemon_action_set_image(self, action: str, image: Optional[str], daemon_typ
             if action != 'redeploy':
                 raise OrchestratorError(
                     f'Cannot execute {action} with new image. `action` needs to be `redeploy`')
-            if daemon_type not in CEPH_TYPES:
+            # need iscsi and nfs as well in order to upgrade them
+            if daemon_type not in CEPH_TYPES and daemon_type not in ['nfs', 'iscsi']:
```

Comment:
this hunk is still needed?

Reply:
What do you mean by this comment? Do you think we can just remove the check and allow custom images for any daemon type?